### ISSUE TITLE
More about traefik?

### ISSUE THREAD

**Issue by gerroon** [norweeg/self-hosted-docker-server-templates#2]
Hi

Can you please document how you tie Traefik in these setups? There is no real documentation on how you do it.

thanks

**Comment by norweeg** [comment]
hi @gerroon!  Is there something specific you were curious about regarding traefik?  Are you asking how does traefik know what and how to proxy requests or are you asking how the traefik config.toml file works?  Something else?  I can definitely add something, I just need to know what exactly you're looking for.  When I was working on these templates, I was referring heavily on the following pages of traefik's documentation:

- [How the Traefik.toml file works](https://docs.traefik.io/configuration/commons/).  This includes how Traefik is configured to listen on a specific port, how to redirect a request from one to another (i.e. http to https) and whether or not these ports use encryption (see entrypoints section) and how to configure the automatic obtaining of encryption certificates  (see ACME section)
- [How Traefik can be told to get configuration options via docker labels to automatically configure new frontends/backends when a container is created](https://docs.traefik.io/configuration/backends/docker/)  This includes a complete list of every label you can add to a container to tell Traefik when and how to forward requests to it, or if it should forward requests to it at all.

**Comment by gerroon** [comment]
I was thinking that a sample toml and docker sample with one of the templates you have that you know it works. Otherwise it becomes  a bit of tutorial hunting

**Comment by norweeg** [comment]
@gerroon is [configs/traefik.toml](https://github.com/norweeg/self-hosted-docker-server-templates/blob/master/configs/traefik.toml) insufficient?  I'm not sure what else I can do besides enumerate the meaning of each section/option, but that would just be [duplicating traefik's on documentation on the file format](https://docs.traefik.io/basics/).  I could maybe add a header that links the original documentation, but I'd rather avoid re-enumerating and repeating the functions of each section/option, especially when the original documentation already thoroughly documents it with examples and has a page of [a small cookbook of examples](https://docs.traefik.io/user-guide/examples/) (which I used extensively, myself, when writing this), but if there's something I can clarify just direct me to what and where.  

As for the docker bit, I did add annotations to the label sections in [docker-compose.base.yaml](https://github.com/norweeg/self-hosted-docker-server-templates/blob/master/docker-compose.base.yaml) explaining what they did, but opted to not redundantly annotate the same types of lines in each file.  Once you've read the base file, with few exceptions, you've read them all.  I *believe* I annotated lines when I used new traefik docker labels on a container that I had not used in the base compose yaml file, explaining what they are and why they're there, but I'm not 100% confident that I did everywhere I may have done that.  Is there a specific compose file/label you want me to revisit and  annotate??

In short, can you help me by narrowing the scope of your question to specifically what you want to know/have me annotate so I don't end up needlessly repeating myself or the traefik documentation, or if you're having difficulty setting this up for yourself, can you tell me what you're getting stuck on and I can help?

**Comment by gerroon** [comment]
Ok both yaml and docker annotations are useful, if you could mention those additional bits in the read me that would be great so the users might know where to look for those.

thanks.

**Comment by norweeg** [comment]
@gerroon will do!  thanks for the suggestion!  I'll add them shortly

**Comment by norweeg** [comment]
@gerroon added your requested section and went back and added some annotations to notable instances where new/different traefik labels were used.  Hope this helps!

**Comment by gerroon** [comment]
thanks
